pkg/processor/metricfrequencyprocessor/sieve.go

```diff
@@ -0,0 +1,197 @@
+package metricfrequencyprocessor
+
+import (
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"math"
+	"sort"
+	"time"
+)
+
+const (
+	float64EqualityThreshold = 1e-9
+	minPointAccumulationTime = time.Minute * 15
```

Comment:
A common pattern is to make it configurable but always start with default values